File: java/src/test/java/route/TargetTest.java
package route;

import static org.junit.Assert.*;
import org.json.JSONObject;
import org.junit.Before;
import org.junit.Test;

public class TargetTest {

    private Driver driver;
    private JSONObject target = new JSONObject("{8631: 'Mr'," +
            "8630: 'Test Java'," +
            "8628: 'testJAVA@test.com'}");
    private JSONObject targetUpdate = new JSONObject("{8631: 'Mr'," +
            "8630: 'Test Java Update'," +
            "8628: 'test@test.com'}");
    private String id = "000OG3XX";

    @Before
    public void setUp() {
        JSONObject config = new JSONObject();

        config.put("xkey", "my_xkey");
        config.put("url", "http://127.0.0.1:8888/");
        driver = new Driver(config);
    }

    @Test
    public void testCreate() {
        JSONObject response = this.driver.target.create(this.target);
        assertEquals(response.getInt("statusCode"), 200);
    }

    @Test
    public void testUpdate() {
        JSONObject response = this.driver.target.update(this.id, this.targetUpdate);
        assertEquals(response.getInt("statusCode"), 200);
    }

    @Test
    public void testGet() {
        JSONObject response = this.driver.target.get(this.id);
        assertEquals(response.getInt("statusCode"), 200);
    }

    @Test
    public void testList() {
        JSONObject response = this.driver.target.list();
        assertEquals(response.getInt("statusCode"), 200);
    }

    @Test
    public void testFind() {
        JSONObject response = this.driver.target.find("test@test.com");
        assertEquals(response.getInt("statusCode"), 200);
    }

    @Test
    public void testDelete() {
        JSONObject response = this.driver.target.delete(this.id);
        assertEquals(response.getInt("statusCode"), 204);
    }
}

File: csharp/DriverTest/TargetTest.cs
using System.Collections.Generic;
using System.Net;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using RestSharp;

namespace DriverTest
{
    [TestClass]
    public class TargetTest
    {
        private static NP6Api.Driver _driver;

        [ClassInitialize()]
        public static void TargetTestInitialize(TestContext testContext)
        {
            var config = new Dictionary<string, string>();

            config["xKey"] = "YOUR XKEY";
            config["url"] = "http://127.0.0.1:8080/";
            _driver = new NP6Api.Driver(config);
        }

        [TestMethod]
        public void TestCreate()
        {
            var data = new JsonObject
            {
                {"1234", "test@test.com"},
                {"1234", "Test"}, 
                {"1234", "Mr"}
            };

            var response = _driver.Target.Create(data);

            Assert.AreEqual((HttpStatusCode)200, response.StatusCode);
        }

        [TestMethod]
        public void TestUpdate()
        {
            var data = new JsonObject
            {
                {"1234", "test@test.com"},
                {"1234", "Test"}, 
                {"1234", "Mr"}
            };

            var response = _driver.Target.Update("01234ABC", data);

            Assert.AreEqual((HttpStatusCode)200, response.StatusCode);
        }

        [TestMethod]
        public void TestGet()
        {
            var response = _driver.Target.Get("01234ABC");

            Assert.AreEqual((HttpStatusCode)200, response.StatusCode);
        }

        [TestMethod]
        public void TestList()
        {
            var response = _driver.Target.List();

            Assert.AreEqual((HttpStatusCode)200, response.StatusCode);
        }

        [TestMethod]
        public void TestFind()
        {
            var response = _driver.Target.Find("test@np6.com");
            
            Assert.AreEqual((HttpStatusCode)200, response.StatusCode);
        }

        [TestMethod]
        public void TestDelete()
        {
            var response = _driver.Target.Delete("01234ABC");

            Assert.AreEqual((HttpStatusCode)204, response.StatusCode);
        }
    }
}
